File: python_2026/gg1/matma.cc
#include <iostream>
#include <cmath>
#include <chrono>
#include <iomanip>

/**
* Wersja C++ programu do liczenia sin (matma.py).
* Kompilacja przez g++ -O3 matma.cc
*/

double get_sin(double degree) {
    return std::sin(degree * M_PI / 180.0);
}

int main() {
    double x = 0.0;
    auto start = std::chrono::high_resolution_clock::now();

    for (int i = 0; i < 10000000; ++i) {
        x += get_sin(i);
    }


    auto end = std::chrono::high_resolution_clock::now();
    std::chrono::duration<double> duration = end - start;

    std::cout << "duration: " << std::fixed << std::setprecision(6)
              << duration.count() << std::endl;
    std::cout << x << std::endl;


    return 0;
}

File: python_2026/gg1/matma.py
import math
from datetime import datetime


def ts() -> float:
    return datetime.now().timestamp()


def duration(start_ts) -> float:
    return ts() - start_ts

def get_sin(degree: float) -> float:
    return math.sin(degree * math.pi / 180)


if __name__ == '__main__':
    x = 0
    st = ts()
    for i in range(10 ** 6):
        x += get_sin(i)

    print(f'duration: {duration(st):.6f}')